### PR TITLE
Update Audio Device icon when switching Audio device

### PR DESCRIPTION
**Pull request by phileo99**
## Purpose

* Fix a bug on setup and call screen where changing audio device does not change icon

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

* Test the code

```
In Lobby screen:
* tap audio device
* select "Speaker"

verify audio device icon has not changed

* tap audio device
* select Android

verify audio device icon has changed

* place app into background, then return to foreground

verify audio device icon has not changed

* Repeat test for Call screen
```

### CODE REVIEW

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java [hunk 1]
```diff
@@ -82,6 +86,17 @@
...
     private ParticipantListPopupWindow participantListPopupWindow;
+    private ImageButton deviceOptionsButton;
+
+    private final Consumer<AudioDeviceType> audioDeviceTypeConsumer = new Consumer<AudioDeviceType>() {
+        @Override
+        public void accept(final AudioDeviceType audioDeviceType) {
+            final Drawable audioDevice = ContextCompat.getDrawable(CallActivity.this,
+                    audioDeviceType == AudioDeviceType.ANDROID
+                            ? R.drawable.ic_fluent_speaker_2_28_filled : R.drawable.ic_fluent_speaker_2_28_regular);
+            deviceOptionsButton.setImageDrawable(audioDevice);
```
Comment (Gemakk) [inline AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java:97]: Does this get run in the UI thread?

**Comment by lpx1989** [comment]
As the screenshot shows, after rotation, the updated icon is reset back to previous device.

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java (hunk 1 above)
Comment (phileo99) [reply AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java]: Yes, confirmed with debugger that `audioDeviceTypeConsumer.accept()` is run in ActivityThread.java, which is the Main thread loop

**Comment by phileo99** [comment]
> 
> As the screenshot shows, after rotation, the updated icon is reset back to previous device.

Good catch @lpx1989 , I have committed a fix

**Comment by iaulakh** [comment]
@phileo99 I noticed that icon updated as expected but text below icon is not updated.
Screen Shot 2021-11-15 at 7 44 22 PM

**Comment by phileo99** [comment]
The icon in calling screen now matches what was selected in the device selection bottom sheet
Screenshot_20211215-125556_Azure Calling

Screenshot_20211215-123501_Azure Calling

File: AzureCalling/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -52,7 +52,7 @@
         <activity
             android:name="com.azure.samples.communication.calling.activities.CallActivity"
             android:exported="true"
-            android:configChanges="orientation|screenSize|screenLayout"
+            android:configChanges="screenSize|screenLayout"
```
Comment (iaulakh) [inline AzureCalling/app/src/main/AndroidManifest.xml:55]: @phileo99 do we need need orientation here?

Comment (phileo99) [reply AzureCalling/app/src/main/AndroidManifest.xml]: @iaulakh you're right, we are implementing `onConfigurationChanged()`.  I have added that configChange back in

**Review by ShaunaSong — APPROVED** [review]

**Review by lpeixin — APPROVED** [review]

**Pull request merged**